src/lib/parser_impl.c

```diff
@@ -396,32 +396,41 @@ __Z_INLINE parser_error_t _readTx(parser_tx_t *v, CborValue *it) {
 
     MEMZERO(&v->oasis.tx, sizeof(oasis_tx_t));
 
-    CHECK_CBOR_TYPE(cbor_value_get_type(it), CborMapType);
-    // FIXME: expected count can be 2 or 3 or 4 see #17
-    //CHECK_CBOR_MAP_LEN(&it, 4);
+    uint8_t valuesCount = 0;
 
+    CHECK_CBOR_TYPE(cbor_value_get_type(it), CborMapType);
 
     // Find method and read it first
     CborValue methodField;
     CHECK_CBOR_ERR(cbor_value_map_find_value(it, "method", &methodField));
     CHECK_PARSER_ERR(_readMethod(v, &methodField));
+    valuesCount++;
 
     // FIXME: fee is optional see #17
     CborValue feeField;
     CHECK_CBOR_ERR(cbor_value_map_find_value(it, "fee", &feeField));
-    CHECK_PARSER_ERR(_readFee(v, &feeField));
+    // We have fee
+    if (cbor_value_is_valid(&feeField)) {
```

Comment:
When we don't have fee, we should have some boolean = false and later skip the fee in the UI too